nemo/collections/common/losses/cross_entropy.py

```diff
@@ -0,0 +1,81 @@
+# =============================================================================
+# Copyright 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import torch
+from torch import nn
+
+from nemo.core.classes import Serialization, Typing, typecheck
+from nemo.core.neural_types import LabelsType, LengthsType, LogitsType, LogprobsType, LossType, MaskType, NeuralType
+from nemo.utils.decorators import experimental
+
+__all__ = ['CrossEntropyLoss']
+
+
+@experimental
+class CrossEntropyLoss(nn.CrossEntropyLoss, Serialization, Typing):
+    """
+    CrossEntropyLoss
+    Args:
+        logits_ndim (int): number of dimensions (or rank) of the logits tensor
+        weight (list): list of rescaling weight given to each class
+        reduction (str): type of the reduction over the batch
+    """
+
+    @property
+    def input_ports(self):
+        """Returns definitions of module input ports.
+        """
+        return {
+            "logits": NeuralType(['B'] + ['ANY'] * (self._logits_dim - 1), LogitsType()),
+            "labels": NeuralType(['B'] + ['ANY'] * (self._logits_dim - 2), LabelsType()),
+            "loss_mask": NeuralType(['B'] + ['ANY'] * (self._logits_dim - 2), MaskType(), optional=True),
+        }
+
+    @property
+    def output_ports(self):
+        """Returns definitions of module output ports.
+        loss:
+            NeuralType(None)
+        """
+        return {"loss": NeuralType(elements_type=LossType())}
+
+    def __init__(self, logits_ndim=2, weight=None, reduction='mean'):
+        super().__init__(weight=weight, reduction=reduction)
```

Comment:
could you please add docstrings/hints here, there was some confusion from the users about the logits_ndim

Reply:
ok